lib/binance/futures.ex

```diff
@@ -0,0 +1,372 @@
+defmodule Binance.Futures do
+  alias Binance.Futures.Rest.HTTPClient
+
+  @type error ::
+          {:binance_error, %{code: integer(), message: String.t()}}
+          | {:http_error, any()}
+          | {:poison_decode_error, any()}
+          | {:config_missing, String.t()}
+
+  # Server
+
+  @doc """
+  Pings Binance API. Returns `{:ok, %{}}` if successful, `{:error, reason}` otherwise
+  """
+  @spec ping() :: {:ok, %{}} | {:error, error()}
+  def ping() do
+    HTTPClient.get_binance("/fapi/v1/ping")
+  end
+
+  @doc """
+  Get binance server time in unix epoch.
+
+  ## Example
+  ```
+  {:ok, 1515390701097}
+  ```
+
+  """
+  @spec get_server_time() :: {:ok, integer()} | {:error, error()}
+  def get_server_time() do
+    case HTTPClient.get_binance("/fapi/v1/time") do
+      {:ok, %{"serverTime" => time}} -> {:ok, time}
+      err -> err
+    end
+  end
+
+  @spec get_exchange_info() :: {:ok, %Binance.ExchangeInfo{}} | {:error, error()}
+  def get_exchange_info() do
+    case HTTPClient.get_binance("/fapi/v1/exchangeInfo") do
+      {:ok, data} -> {:ok, Binance.ExchangeInfo.new(data)}
+      err -> err
+    end
+  end
+
+  @spec get_symbol_price :: {:ok, list(%Binance.SymbolPrice{})} | {:error, error()}
+  def get_symbol_price() do
+    case HTTPClient.get_binance("/fapi/v1/ticker/price") do
+      {:ok, data} -> {:ok, Enum.map(data, &Binance.SymbolPrice.new/1)}
+      err -> err
+    end
+  end
+
+  @spec create_listen_key(map()) :: {:ok, map()} | {:error, error()}
+  def create_listen_key(
+        receiving_window \\ 1500,
+        timestamp \\ nil,
+        config \\ nil
+      ) do
+    timestamp =
+      case timestamp do
+        # timestamp needs to be in milliseconds
+        nil ->
+          :os.system_time(:millisecond)
+
+        t ->
+          t
+      end
+
+    arguments = %{
+      timestamp: timestamp,
+      recvWindow: receiving_window
+    }
+
+    case HTTPClient.post_binance("/fapi/v1/listenKey", arguments, config) do
+      {:ok, %{"code" => code, "msg" => msg}} ->
+        {:error, {:binance_error, %{code: code, msg: msg}}}
+
+      data ->
+        data
+    end
+  end
+
+  @spec keep_alive_listen_key(integer(), integer() | nil, map() | nil) ::
+          {:ok, %{}} | {:error, error()}
+  def keep_alive_listen_key(receiving_window \\ 1500, timestamp \\ nil, config \\ nil) do
+    timestamp =
+      case timestamp do
+        # timestamp needs to be in milliseconds
+        nil ->
+          :os.system_time(:millisecond)
```

Comment:
There is already a function in BinanceHelper that does this.

Maybe this whole `case` could be extracted into a function since it's used in a lot of places